src/data-types.js

```diff
@@ -518,18 +518,18 @@ class DATE extends ABSTRACT {
 
   _applyTimezone(date, options) {
     if (options.timezone) {
-      if (momentTz.tz.zone(options.timezone)) {
-        return momentTz(date).tz(options.timezone);
+      if (isValidTimeZone(options.timezone)) {
+        return dayjs(date).tz(options.timezone);
       }
 
-      return moment(date).utcOffset(options.timezone);
+      return dayjs(date).utcOffset(options.timezone);
     }
 
-    return momentTz(date);
+    return dayjs(date);
   }
 
   _stringify(date, options) {
-    if (!moment.isMoment(date)) {
+    if (!dayjs.isDayjs(date)) {
```

Comment:
This is the only line that worries me.
Doesn't this mean that, if `options.timezone` is set, then all dates will have their timeZone changed, except if it's an instance of dayjs

I think we should either always overwrite the timeZone (including dayjs)

Edit: it looks like that was the initial intention, as [a recent PR](https://github.com/sequelize/sequelize/issues/13816) introduced this change. I'll look into it more

Reply:
Okay 👍

Reply:
> This is the only line that worries me. Doesn't this mean that, if `options.timezone` is set, then all dates will have their timeZone changed, except if it's an instance of dayjs
> 
> I think we should either always overwrite the timeZone (including dayjs)
> 
> Edit: it looks like that was the initial intention, as [a recent PR](https://github.com/sequelize/sequelize/pull/13818) introduced this change. I'll look into it more, see if that change should not be reverted.

I can help you to explain the change, let's checkout to that PR.

Reply:
After the discussion in the other thread, I think it's safe to remove the condition, and call _applyTimezone to all inputs

Reply:
Okay 👍